crypto/engine/eng_ctrl.c

```diff
@@ -8,6 +8,7 @@
  */
 
 #include "eng_int.h"
+#include <stdio.h>
```

Comment:
I'm curious, how come we have had *no* warnings of implicit declaration of `sprintf`?

Reply:
the sprintf calls all seem to use "%s" as format string.  They could be replaced with strcpy and strlen for the
return value.

Reply:
Various of our public header files (including cryptlib.h which is transitively included from this file) have this:

````
# ifndef OPENSSL_NO_STDIO
#  include <stdio.h>
# endif
````

So, normally, we include stdio.h. It's only if you request no-stdio where you have a problem.

Reply:
Hmmm...  `include/openssl/srp.h` doesn't guard the inclusion of `stdio.h` accordingly...  maybe that should be changed to conform with the rest of the public header files?

Also, I'm noticing that `include/openssl/err.h` guards the inclusion of `stdlib.h` with `OPENSSL_NO_STDIO`...  seems odd to me.

Reply:
Also, I agree with @bernd-edlinger, the sprintfs here should be replaced with `strcpy`, or even better, `OPENSSL_strlcpy`.

Reply:
... on second thought, `OPENSSL_strlcpy` may be too much to ask in this case